payment/app.py

```diff
@@ -12,39 +12,164 @@
                               password=os.environ['REDIS_PASSWORD'],
                               db=int(os.environ['REDIS_DB']))
 
-
 def close_db_connection():
     db.close()
 
+gateway_url = os.environ['GATEWAY_URL']
 
 atexit.register(close_db_connection)
 
+class User:
+    def __init__(self, user_id: str):
+        self.user_id = user_id
+        self.credit = 0
+    
+    def to_dict(self):
+        return {
+            "user_id" : self.user_id,
+            "credit" : self.credit,
+        }
 
 @app.post('/create_user')
 def create_user():
-    pass
+
+    # Create new user id
+    new_user_id = f"item:{random.getrandbits(ID_BYTES_SIZE)}"
+    while db.hget(new_user_id, "item_id"):
+        new_user_id = f"item:{random.getrandbits(ID_BYTES_SIZE)}"
+    
+    new_user = User(user_id=new_user_id)
+    
+    try:
+        db.hset(new_user.user_id, mapping=new_user.to_dict())
+    except Exception as exp:
+        return Response(str(exp), status=400)
+
+    return Response(json.dumps(new_user.to_dict()), mimetype="application/json", status=200)
 
 
 @app.get('/find_user/<user_id>')
 def find_user(user_id: str):
-    pass
+    
+    user = db.hgetall(user_id)
+
+    if not user:
+        return Response(f"The user {item_id} does not exist in the DB!", status=404)
+    
+    return_item = {
+        "user_id" : user[b"user_id"].decode("utf-8")
+        "credit" : int(user[b"credit"])
+    }
+    return Response(return_item, mimetype="application/json", status=200)
 
```

Comment:
When testing this endpoint I don't get the correct output. Can you check that?

Reply:
I think this is because of the wrong name when creating a new user.
I will commit to the new change and hope it fixes it.